Sources/WeaverDI/ThreadSafeInstanceCachingDecorator.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  ThreadSafeInstanceCachingDecorator.swift
+//  WeaverDI
+//
+//  Created by Bruno Mazzo on 5/31/18.
+//
+
+import Foundation
+
+class ThreadSafeInstanceCachingDecorator: InstanceCaching {
+    
+    let cache: InstanceCaching
```

Comment:
Let's rename this to `instances` to be consistent with the rest of the codebase

Reply:
Done